src/util/OAuth2Util.js

```diff
@@ -16,34 +16,58 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
   var _ = Okta._;
 
   /**
-   * Get the 'id_token' from a social/okta idps.
+   * Get the tokens in the OIDC/OAUTH flows
    *
    * @param settings - settings model object
    * @param params - {idp: 'xxx'} for social auth
    *                 {sessionToken: 'xxx'} for okta idp
    */
-  util.getIdToken = function (settings, params) {
+  util.getTokens = function (settings, params) {
+
+    function success(result) {
+      settings.callGlobalSuccess(Enums.SUCCESS, result);
+    }
+
+    function error(error) {
+      settings.callGlobalError(new Errors.OAuthError(error.message));
+    }
+
     var authClient = settings.getAuthClient(),
-        options = settings.toJSON({verbose: true}),
-        oauthParams = _.extend({}, _.pick(options, 'clientId'),
-          _.pick(options.authParams, 'responseType', 'responseMode', 'display', 'scope', 'scopes')),
+        options = settings.toJSON({ verbose: true }),
+        oauthParams = {},
         extraOptions = {};
 
-    _.extend(oauthParams, params || {}, {redirectUri: options.oauthRedirectUri});
+    _.extend(
+      oauthParams,
+      _.pick(options, 'clientId', 'redirectUri'),
+      _.pick(options.authParams, 'responseType', 'responseMode', 'display', 'scope', 'scopes'),
+      params
+    );
 
     // Extra Options for Social Idp popup window title and id_token response timeout
     extraOptions.popupTitle = Okta.loc('socialauth.popup.title', 'login');
-    if (options.oAuthTimeout) {
-      extraOptions.timeOut = options.oAuthTimeout;
+    extraOptions.timeOut = options.oAuthTimeout;
```

Comment:
this should be `timeout` - https://github.com/okta/okta-auth-js/blob/master/lib/token.js#L434